pkg/apis/stork/v1alpha1/applicationclone.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	//ApplicationCloneResourceName is the name for the application clone resource
+	ApplicationCloneResourceName = "applicationclone"
+	//ApplicationCloneResourcePlural is the name in plural for the application clone resources
+	ApplicationCloneResourcePlural = "applicationclones"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//ApplicationClone represents the cloning of application in different namespaces
+type ApplicationClone struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            ApplicationCloneSpec   `json:"spec"`
+	Status          ApplicationCloneStatus `json:"status,omitempty"`
+}
+
+//ApplicationCloneSpec defines the spec to create an application clone
+type ApplicationCloneSpec struct {
+	//NamespaceMapping to store the target:destination namespaces
+	NamespaceMapping map[string]string `json:"namespace_mapping"`
+	//Selectors for label on objects
+	Selectors    map[string]string `json:"selectors"`
+	PreExecRule  string            `json:"preExecRule"`
+	PostExecRule string            `json:"postExecRule"`
+}
+
+//ApplicationCloneStatus defines the status of the clone
+type ApplicationCloneStatus struct {
+	//Status of the cloning process
```

Comment:
We need to add more info here about what volumes and resources are being cloned and the status for each, similar to the Migration spec. It'll help users know what was successful and what failed.

Reply:
I have added volumeInfo and resourceInfo please check and let me know if we need to add something more to the info structs.